batchflow/models/tf/mobilenet.py

```diff
@@ -189,38 +232,112 @@ def body(cls, inputs, name='body', **kwargs):
                 for k in range(repeats):
                     if k > 0:
                         block['strides'] = 1
-                    x = cls.block(x, **block, residual=k > 0, name='block-%d' % i, **kwargs)
+                    residual = block.pop('residual', False) or k > 0
+                    x = cls.block(x, **block, residual=residual, name='block-%d' % i, **kwargs)
                     i += 1
         return x
 
     @classmethod
-    def block(cls, inputs, filters, residual=False, strides=1, expansion_factor=6, width_factor=1, name=None, **kwargs):
-        """ A network building block consisting of a separable depthwise convolution and 1x1 pointwise covolution.
+    def block(cls, inputs, filters, residual=False, strides=1, expansion_factor=4, width_factor=1, kernel_size=3,
+              se_block=False, name=None, **kwargs):
+        """ An inverted residual bottleneck block consisting of a separable depthwise convolution and 1x1 pointise
+        convolution and an optional Squeeze-and-Excitation block.
 
         Parameters
         ----------
         inputs : tf.Tensor
             input tensor
+        repeats : int
+            number of repeats of the block
+        kernel_size : int
+            kernel size for depthwise convolution
         filters : int
             number of output filters
-        residual : boold
+        residual : bool
             whether to make a residual connection
         strides : int
             stride for 3x3 convolution
         expansion_factor : int
             multiplier for the number of filters in internal convolutions
         width_factor : float
             multiplier for the number of filters
+        activation : callable, optional
+            If not specified tf.nn.relu is used.
+        se_block : bool
+            whether to include squeeze and excitation block
         name : str
             scope name
 
         Returns
         -------
         tf.Tensor
         """
+
         num_filters = int(cls.num_channels(inputs, kwargs.get('data_format')) * expansion_factor * width_factor)
-        conv_filters = [num_filters, num_filters, filters]
-        x = conv_block(inputs, 'cna Cna cn', conv_filters, [1, 3, 1], name=name, strides=[1, strides, 1], **kwargs)
+        x = conv_block(inputs, 'cnawna', [num_filters, num_filters], [1, kernel_size], strides=[1, strides],
+                       name='%s-exp' % name, **kwargs)
+
+        if se_block:
+            if not isinstance(se_block, dict):
+                se_block = dict(activation=[kwargs.get('activation', tf.nn.relu), h_sigmoid], ratio=num_filters // 4)
+            x = cls.se_block(x, **se_block, name='%s-se' % name, data_format=kwargs.get('data_format'))
```

Comment:
pass all kwargs, not only data format

Reply:
then if there is activation in kwargs, it has to be removed first and added back for the next conv_block.

Reply:
fixed.